hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/AbstractNotificationView.java

```diff
@@ -0,0 +1,165 @@
+
+    private transient Map<Class<?>, RefreshableContainer> supportedEvents;
+
+    private final NotificationUnreadButton notificationUnreadButton;
+
+    private int viewUnreadNotifcations;
+
+    /**
+     * Constructor.
+     * 
+     * @param eventBus
+     *            the ui event bus
+     * @param notificationUnreadButton
+     *            the notificationUnreadButton
+     */
+    public AbstractNotificationView(final EventBus.UIEventBus eventBus,
+            final NotificationUnreadButton notificationUnreadButton) {
+        this.eventBus = eventBus;
+        this.notificationUnreadButton = notificationUnreadButton;
+    }
+
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onEventContainerEvent(final EventContainer<?> eventContainer) {
+        if (!supportNotificationEventContainer(eventContainer.getClass()) || eventContainer.getEvents().isEmpty()) {
+            return;
+        }
+
+        eventContainer.getEvents().stream().filter(event -> !anyEventMatch(event)).forEach(event -> {
+            notificationUnreadButton.incrementUnreadNotification(this, eventContainer);
+            viewUnreadNotifcations++;
+        });
+        getDashboardMenuItem().setNotificationUnreadValue(viewUnreadNotifcations);
+    }
+
+    private boolean anyEventMatch(final TenantAwareEvent tenantAwareEvent) {
+        return skipUiEvents.asMap().keySet().stream().anyMatch(uiEvent -> uiEvent.matchRemoteEvent(tenantAwareEvent));
+    }
+
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onUiEvent(final BaseUIEntityEvent<?> event) {
+        if (BaseEntityEventType.ADD_ENTITY != event.getEventType()
+                && BaseEntityEventType.REMOVE_ENTITIES != event.getEventType()
+                && BaseEntityEventType.UPDATED_ENTITY != event.getEventType()) {
+            return;
+        }
+        skipUiEvents.put(event, new Object());
+    }
+
+    @PostConstruct
+    protected void subscribe() {
+        eventBus.subscribe(this);
+    }
+
+    @PreDestroy
+    protected void destroy() {
+        eventBus.unsubscribe(this);
+    }
+
+    /**
+     * Refresh the view by event container changes.
+     * 
+     * @param eventContainers
+     *            event container which container changed
+     * 
+     */
+    public void refreshView(final Set<Class<?>> eventContainers) {
+        eventContainers.stream().filter(clazz -> supportNotificationEventContainer(clazz))
+                .forEach(clazz -> refreshContainer(clazz));
+        clear();
+    }
+
+    protected void refreshContainer(final Class<?> containerClazz) {
+        getSupportedEvents().get(containerClazz).refreshContainer();
+    }
+
+    /**
+     * Refresh the view by event container changes.
+     * 
+     * 
+     */
+    public void refreshView() {
+        if (viewUnreadNotifcations <= 0) {
+            return;
+        }
+        supportedEvents.values().stream().forEach(container -> container.refreshContainer());
```

Comment:
`supportedEvents` might be null in case I call `refreshView()` before `getSupportedEvents()` is called